src/query/service/src/interpreters/interpreter_copy.rs

```diff
@@ -246,7 +246,13 @@ impl CopyInterpreter {
         }
 
         let mut stage_table_info = stage_table_info.clone();
-        let mut all_source_file_infos = StageTable::list_files(&stage_table_info).await?;
+        let max_files = stage_table_info.stage_info.copy_options.max_files;
```

Comment:
If max_files is zero, we can set to `usize::max`, the following code do not need to check zero now.